detectron2/checkpoint/model_zoo.py

```diff
@@ -1,132 +1,60 @@
 # Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
-import logging
-from fvcore.common.file_io import PathHandler, PathManager
+import os  # noqa: F401
+import pkg_resources  # noqa: F401
+from typing import Dict, List
 
+from detectron2.config import get_cfg  # noqa: F401
+from detectron2.modeling import build_model  # noqa: F401
+from .catalog import Detectron2ModelCatalog
+from .detection_checkpoint import DetectionCheckpointer  # noqa: F401
 
-class ModelCatalog(object):
-    """
-    Store mappings from names to third-party models.
-    """
 
-    S3_C2_DETECTRON_PREFIX = "https://dl.fbaipublicfiles.com/detectron"
+_SUPPORTED_TASKS: Dict[str, List[str]] = {}
 
-    # MSRA models have STRIDE_IN_1X1=True. False otherwise.
-    # NOTE: all BN models here have fused BN into an affine layer.
-    # As a result, you should only load them to a model with "FrozenBN".
-    # Loading them to a model with regular BN or SyncBN is wrong.
-    # Even when loaded to FrozenBN, it is still different from affine by an epsilon,
-    # which should be negligible for training.
-    # NOTE: all models here uses PIXEL_STD=[1,1,1]
-    C2_IMAGENET_MODELS = {
-        "MSRA/R-50": "ImageNetPretrained/MSRA/R-50.pkl",
-        "MSRA/R-101": "ImageNetPretrained/MSRA/R-101.pkl",
-        "FAIR/R-50-GN": "ImageNetPretrained/47261647/R-50-GN.pkl",
-        "FAIR/R-101-GN": "ImageNetPretrained/47592356/R-101-GN.pkl",
-        "FAIR/X-101-32x8d": "ImageNetPretrained/20171220/X-101-32x8d.pkl",
-        "FAIR/X-101-64x4d": "ImageNetPretrained/FBResNeXt/X-101-64x4d.pkl",
-        "FAIR/X-152-32x8d-IN5k": "ImageNetPretrained/25093814/X-152-32x8d-IN5k.pkl",
-    }
+MODEL_ZOO_FUNCTION_STRING = """
```

Comment:
rather than using exec and string manipulation, you can use a function creator which takes some arguments and returns a function to be used as model zoo function.

Reply:
Great idea! Making this modification.